pandas/tests/test_frame.py

```diff
@@ -3568,6 +3565,20 @@ def test_constructor_tuples(self):
         expected = DataFrame({'A': Series([(1, 2), (3, 4)])})
         assert_frame_equal(result, expected)
 
+    def test_constructor_namedtuples(self):
+        # GH11181
+        from collections import namedtuple
+        named_tuple = namedtuple("Pandas", list('ab'))
+        tuples = [named_tuple(1, 3), named_tuple(2, 4)]
+        expected = DataFrame({'a': [1, 2], 'b': [3, 4]})
+        result = DataFrame(tuples)
+        assert_frame_equal(result, expected)
+
```

Comment:
I think the correct way would be to compare the columns with *each* of the namedtuples and if they differ then raise a ``ValueError``. This *might* be expensive, so what I would do instead is keep track of the type of namedtuple. and instead just compare that. If they differ then easiest to just raise a ``ValueError`` (in theory you could just discard ``columns`` at this point, but I think this is an actual error).